---
int32 handle
int32 a
int32 b
int32 rj
